src/Controller/PerformanceBudgetController.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\performance_budget\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueWorkerManagerInterface;
+use Drupal\Core\Queue\RequeueException;
+use Drupal\performance_budget\Entity\PerformanceBudget;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class PerformanceBudgetController.
+ *
+ * @package Drupal\performance_budget\Controller
+ */
+class PerformanceBudgetController extends ControllerBase {
+
+  /**
+   * Drupal\Core\Queue\QueueFactory definition.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queue;
+
+  /**
+   * Drupal\Core\Queue\QueueWorkerManagerInterface definition.
+   *
+   * @var \Drupal\Core\Queue\QueueWorkerManagerInterface
+   */
+  protected $queueManager;
+
+  /**
+   * Constructs a new WebPageArchiveController object.
+   */
+  public function __construct(QueueFactory $queue, QueueWorkerManagerInterface $queue_manager) {
+    $this->queue = $queue;
+    $this->queueManager = $queue_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('queue'),
+      $container->get('plugin.manager.queue_worker')
+    );
+  }
+
+  /**
+   * Returns render array for displaying run history.
+   */
+  public function viewRuns($web_page_archive) {
```

Comment:
Does it need passing at all?  Not that the theme hook is registered but if it's _all runs_ then it doesn't need anything to base it from?